jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt

```diff
@@ -4,15 +4,43 @@
 package software.aws.toolkits.jetbrains.services.codewhisperer.language
 
 import com.intellij.openapi.components.service
+import software.aws.toolkits.jetbrains.services.codewhisperer.model.ProgrammingLanguage
 import software.aws.toolkits.telemetry.CodewhispererLanguage
 
 class CodeWhispererLanguageManager {
-    fun isLanguageSupported(language: String): Boolean =
-        language == CodewhispererLanguage.Java.toString() ||
-            language == CodewhispererLanguage.Python.toString() ||
-            language == CodewhispererLanguage.Javascript.toString()
+    private val supportedLanguage = setOf(
+        CodewhispererLanguage.Java.toString(),
+        CodewhispererLanguage.Python.toString(),
+        CodewhispererLanguage.Javascript.toString()
+    )
+
+    fun isLanguageSupported(language: ProgrammingLanguage): Boolean {
+        val mappedLanguage = getParentLanguage(language)
+        return supportedLanguage.contains(mappedLanguage.languageName)
+    }
+
+    /**
+     * This should be called to map some language dialect to their mother language
+     * e.g. JSX -> JavaScript, TypeScript -> JavaScript etc.
+     */
+    internal fun getParentLanguage(language: ProgrammingLanguage): ProgrammingLanguage =
+        when {
+            language.languageName.contains("jsx") -> ProgrammingLanguage(CodewhispererLanguage.Javascript)
+            else -> language
+        }
 
     companion object {
         fun getInstance(): CodeWhispererLanguageManager = service()
     }
 }
+
+fun ProgrammingLanguage.toCodeWhispererLanguage() = when {
+    languageName.contains("python") -> CodewhispererLanguage.Python
+    languageName.contains("javascript") -> CodewhispererLanguage.Javascript
+    languageName.contains("java") -> CodewhispererLanguage.Java
+    languageName.contains("jsx") -> CodewhispererLanguage.Jsx
+    languageName.contains("""plain.*text""".toRegex()) -> CodewhispererLanguage.Plaintext
```

Comment:
Why are we having this regex? What are the patterns we are seeing for plain text?

Reply:
actually just let me revert it back to a string literal `plain_text` would be fine as well